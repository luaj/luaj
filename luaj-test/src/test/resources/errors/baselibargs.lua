package.path = "?.lua;test/lua/errors/?.lua"
require 'args'

-- arg types for basic library functions

-- assert
banner('assert')
checkallpass('assert',{{true,123},anylua})
checkallerrors('assert',{{nil,false,n=2},{nil,n=1}},'assertion failed')
checkallerrors('assert',{{nil,false,n=2},{'message'}},'message')

-- collectgarbage
banner('collectgarbage')
checkallpass('collectgarbage',{{'collect','count'}},true)
checkallerrors('collectgarbage',{{astring, anumber}},'bad argument')
checkallerrors('collectgarbage',{{aboolean, atable, afunction, athread}},'string expected')

-- dofile
banner('dofile')
--checkallpass('dofile', {})
--checkallpass('dofile', {{'test/lua/errors/args.lua'}})
--checkallerrors('dofile', {{'foo.bar'}}, 'cannot open foo.bar')
--checkallerrors('dofile', {nonstring}, 'bad argument')

-- error
banner('error')
--checkallerrors('error', {{'message'},{nil,0,1,2,n=4}}, 'message')
--checkallerrors('error', {{123},{nil,1,2,n=3}}, 123)

-- getmetatable
banner('getmetatable')
checkallpass('getmetatable', {notanil})
checkallerrors('getmetatable',{},'bad argument')

-- ipairs
banner('ipairs')
checkallpass('ipairs', {sometable})
checkallerrors('ipairs', {notatable}, 'bad argument')

-- load
banner('load')
checkallpass('load', {somefunction,{nil,astring,n=2}})
checkallpass('load', {{'return'}})
checkallpass('load', {{'return'},{'mychunk'}})
-- FIXME why is the result not printed?
--checkallpass('load', {{'return a ... b'},{'mychunk'}},true)
checkallerrors('load', {somefunction,nonstring}, 'bad argument')
checkallerrors('load', {{nil,aboolean,atable,athread},notastring}, 'bad argument')
checkallerrors('load', {{'return'},nonstring}, 'bad argument')

-- loadfile
banner('loadfile')
checkallpass('loadfile', {{'args.lua'}})
checkallerrors('loadfile', {nonstring}, 'bad argument')

-- next
banner('next')
checkallpass('next', {sometable,somekey})
checkallerrors('next', {notatable,{nil,1,n=2}}, 'bad argument')
checkallerrors('next', {sometable,nonkey}, 'invalid key')

-- pairs
banner('pairs')
checkallpass('pairs', {sometable})
checkallerrors('pairs', {notatable}, 'bad argument')

-- pcall
banner('pcall')
checkallpass('pcall', {notanil,anylua}, true)
checkallerrors('pcall',{},'bad argument')

-- print
banner('print')
checkallpass('print', {}) 
checkallpass('print', {{nil,astring,anumber,aboolean,n=4}}) 

-- rawequal
banner('rawequal')
checkallpass('rawequal', {notanil,notanil})
checkallerrors('rawequal', {}, 'bad argument')
checkallerrors('rawequal', {notanil}, 'bad argument')

-- rawget
banner('rawget')
checkallpass('rawget', {sometable,somekey})
checkallpass('rawget', {sometable,nonkey})
checkallerrors('rawget', {sometable,somenil},'bad argument')
checkallerrors('rawget', {notatable,notakey}, 'bad argument')
checkallerrors('rawget', {}, 'bad argument')

-- rawset
banner('rawset')
checkallpass('rawset', {sometable,somekey,notanil})
checkallpass('rawset', {sometable,nonkey,notanil})
checkallerrors('rawset', {sometable,somenil},'table index is nil')
checkallerrors('rawset', {}, 'bad argument')
checkallerrors('rawset', {notatable,somestring,somestring}, 'bad argument')
checkallerrors('rawset', {sometable,somekey}, 'bad argument')

-- select
banner('select')
checkallpass('select', {{anumber,'#'},anylua})
checkallerrors('select', {notanumber}, 'bad argument')

-- setmetatable
banner('setmetatable')
checkallpass('setmetatable', {sometable, {nil,atable,n=2}})
checkallerrors('setmetatable',{notatable,sometable},'bad argument')
checkallerrors('setmetatable',{sometable,nontable},'bad argument')

-- tonumber
banner('tonumber')
checkallpass('tonumber',{somenumber,{nil,2,10,36,n=4}})
checkallpass('tonumber',{notanil,{nil,n=1}})
checkallpass('tonumber',{somestring,{10}})
checkallerrors('tonumber',{notastring,{10}},'bad argument')
checkallerrors('tonumber',{{nil,afunction,atable,n=3},{2,9,11,36}},'bad argument')
checkallerrors('tonumber',{somenumber,{1,37,atable,afunction,aboolean}},'bad argument')

-- tostring
banner('tostring')
checkallpass('tostring',{{astring,anumber,aboolean}})
checkallpass('tostring',{{atable,afunction,athread}},true)
checkallpass('tostring',{{astring,anumber,aboolean},{'anchor'}})
checkallpass('tostring',{{atable,afunction,athread},{'anchor'}},true)
checkallerrors('tostring',{},'bad argument')

-- type
banner('type')
checkallpass('type',{notanil})
checkallpass('type',{anylua,{'anchor'}})
checkallerrors('type',{},'bad argument')

-- xpcall
banner('xpcall')
checkallpass('xpcall', {notanil,nonfunction},true)
checkallpass('xpcall', {notanil,{function(...)return 'aaa', 'bbb', #{...} end}},true)
checkallerrors('xpcall',{anylua},'bad argument')


