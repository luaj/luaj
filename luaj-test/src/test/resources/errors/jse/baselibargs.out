====== assert ======
--- checkallpass
-		assert(true,nil)	true
-		assert(123,nil)	123
-		assert(true,'abc')	true,'abc'
-		assert(123,'abc')	123,'abc'
-		assert(true,1.25)	true,1.25
-		assert(123,1.25)	123,1.25
-		assert(true,true)	true,true
-		assert(123,true)	123,true
-		assert(true,<table>)	true,<table>
-		assert(123,<table>)	123,<table>
-		assert(true,<function>)	true,<function>
-		assert(123,<function>)	123,<function>
-		assert(true,<thread>)	true,<thread>
-		assert(123,<thread>)	123,<thread>
--- checkallerrors
-		assert(nil,nil)	...assertion failed...
-		assert(false,nil)	...assertion failed...
--- checkallerrors
-		assert(nil,'message')	...message...
-		assert(false,'message')	...message...
====== collectgarbage ======
--- checkallpass
-		collectgarbage('collect')	number
-		collectgarbage('count')	number,number
--- checkallerrors
-		collectgarbage('abc')	...bad argument...
-		collectgarbage(1.25)	...bad argument...
--- checkallerrors
-		collectgarbage(true)	...string expected...
-		collectgarbage(<table>)	...string expected...
-		collectgarbage(<function>)	...string expected...
-		collectgarbage(<thread>)	...string expected...
====== dofile ======
====== error ======
====== getmetatable ======
--- checkallpass
-		getmetatable('abc')	<table>
-		getmetatable(1.25)	
-		getmetatable(true)	
-		getmetatable(<table>)	
-		getmetatable(<function>)	
-		getmetatable(<thread>)	
--- checkallerrors
-		getmetatable()	...bad argument...
====== ipairs ======
--- checkallpass
-		ipairs(<table>)	<function>,<table>,0
--- checkallerrors
-		ipairs(nil)	...bad argument...
-		ipairs('abc')	...bad argument...
-		ipairs(1.25)	...bad argument...
-		ipairs(true)	...bad argument...
-		ipairs(<function>)	...bad argument...
-		ipairs(<thread>)	...bad argument...
====== load ======
--- checkallpass
-		load(<function>,nil)	<function>
-		load(<function>,'abc')	<function>
--- checkallpass
-		load('return')	<function>
--- checkallpass
-		load('return','mychunk')	<function>
--- checkallerrors
-		load(<function>,true)	...bad argument...
-		load(<function>,<table>)	...bad argument...
-		load(<function>,<function>)	...bad argument...
-		load(<function>,<thread>)	...bad argument...
--- checkallerrors
-		load(nil,nil)	...bad argument...
-		load(true,nil)	...bad argument...
-		load(<table>,nil)	...bad argument...
-		load(<thread>,nil)	...bad argument...
-		load(nil,true)	...bad argument...
-		load(true,true)	...bad argument...
-		load(<table>,true)	...bad argument...
-		load(<thread>,true)	...bad argument...
-		load(nil,<table>)	...bad argument...
-		load(true,<table>)	...bad argument...
-		load(<table>,<table>)	...bad argument...
-		load(<thread>,<table>)	...bad argument...
-		load(nil,<function>)	...bad argument...
-		load(true,<function>)	...bad argument...
-		load(<table>,<function>)	...bad argument...
-		load(<thread>,<function>)	...bad argument...
-		load(nil,<thread>)	...bad argument...
-		load(true,<thread>)	...bad argument...
-		load(<table>,<thread>)	...bad argument...
-		load(<thread>,<thread>)	...bad argument...
--- checkallerrors
-		load('return',true)	...bad argument...
-		load('return',<table>)	...bad argument...
-		load('return',<function>)	...bad argument...
-		load('return',<thread>)	...bad argument...
====== loadfile ======
--- checkallpass
-		loadfile('args.lua')	<function>
--- checkallerrors
-		loadfile(true)	...bad argument...
-		loadfile(<table>)	...bad argument...
-		loadfile(<function>)	...bad argument...
-		loadfile(<thread>)	...bad argument...
====== next ======
--- checkallpass
-		next(<table>,'aa')	
--- checkallerrors
-		next(nil,nil)	...bad argument...
-		next('abc',nil)	...bad argument...
-		next(1.25,nil)	...bad argument...
-		next(true,nil)	...bad argument...
-		next(<function>,nil)	...bad argument...
-		next(<thread>,nil)	...bad argument...
-		next(nil,1)	...bad argument...
-		next('abc',1)	...bad argument...
-		next(1.25,1)	...bad argument...
-		next(true,1)	...bad argument...
-		next(<function>,1)	...bad argument...
-		next(<thread>,1)	...bad argument...
--- checkallerrors
-		next(<table>,'abc')	...invalid key...
-		next(<table>,1.25)	...invalid key...
-		next(<table>,true)	...invalid key...
-		next(<table>,<table>)	...invalid key...
-		next(<table>,<function>)	...invalid key...
====== pairs ======
--- checkallpass
-		pairs(<table>)	<function>,<table>
--- checkallerrors
-		pairs(nil)	...bad argument...
-		pairs('abc')	...bad argument...
-		pairs(1.25)	...bad argument...
-		pairs(true)	...bad argument...
-		pairs(<function>)	...bad argument...
-		pairs(<thread>)	...bad argument...
====== pcall ======
--- checkallpass
-		pcall('abc',nil)	boolean,string
-		pcall(1.25,nil)	boolean,string
-		pcall(true,nil)	boolean,string
-		pcall(<table>,nil)	boolean,string
-		pcall(<function>,nil)	boolean
-		pcall(<thread>,nil)	boolean,string
-		pcall('abc','abc')	boolean,string
-		pcall(1.25,'abc')	boolean,string
-		pcall(true,'abc')	boolean,string
-		pcall(<table>,'abc')	boolean,string
-		pcall(<function>,'abc')	boolean
-		pcall(<thread>,'abc')	boolean,string
-		pcall('abc',1.25)	boolean,string
-		pcall(1.25,1.25)	boolean,string
-		pcall(true,1.25)	boolean,string
-		pcall(<table>,1.25)	boolean,string
-		pcall(<function>,1.25)	boolean
-		pcall(<thread>,1.25)	boolean,string
-		pcall('abc',true)	boolean,string
-		pcall(1.25,true)	boolean,string
-		pcall(true,true)	boolean,string
-		pcall(<table>,true)	boolean,string
-		pcall(<function>,true)	boolean
-		pcall(<thread>,true)	boolean,string
-		pcall('abc',<table>)	boolean,string
-		pcall(1.25,<table>)	boolean,string
-		pcall(true,<table>)	boolean,string
-		pcall(<table>,<table>)	boolean,string
-		pcall(<function>,<table>)	boolean
-		pcall(<thread>,<table>)	boolean,string
-		pcall('abc',<function>)	boolean,string
-		pcall(1.25,<function>)	boolean,string
-		pcall(true,<function>)	boolean,string
-		pcall(<table>,<function>)	boolean,string
-		pcall(<function>,<function>)	boolean
-		pcall(<thread>,<function>)	boolean,string
-		pcall('abc',<thread>)	boolean,string
-		pcall(1.25,<thread>)	boolean,string
-		pcall(true,<thread>)	boolean,string
-		pcall(<table>,<thread>)	boolean,string
-		pcall(<function>,<thread>)	boolean
-		pcall(<thread>,<thread>)	boolean,string
--- checkallerrors
-		pcall()	...bad argument...
====== print ======
--- checkallpass

-		print()	
--- checkallpass
nil
-		print(nil)	
abc
-		print('abc')	
1.25
-		print(1.25)	
true
-		print(true)	
====== rawequal ======
--- checkallpass
-		rawequal('abc','abc')	true
-		rawequal(1.25,'abc')	false
-		rawequal(true,'abc')	false
-		rawequal(<table>,'abc')	false
-		rawequal(<function>,'abc')	false
-		rawequal(<thread>,'abc')	false
-		rawequal('abc',1.25)	false
-		rawequal(1.25,1.25)	true
-		rawequal(true,1.25)	false
-		rawequal(<table>,1.25)	false
-		rawequal(<function>,1.25)	false
-		rawequal(<thread>,1.25)	false
-		rawequal('abc',true)	false
-		rawequal(1.25,true)	false
-		rawequal(true,true)	true
-		rawequal(<table>,true)	false
-		rawequal(<function>,true)	false
-		rawequal(<thread>,true)	false
-		rawequal('abc',<table>)	false
-		rawequal(1.25,<table>)	false
-		rawequal(true,<table>)	false
-		rawequal(<table>,<table>)	true
-		rawequal(<function>,<table>)	false
-		rawequal(<thread>,<table>)	false
-		rawequal('abc',<function>)	false
-		rawequal(1.25,<function>)	false
-		rawequal(true,<function>)	false
-		rawequal(<table>,<function>)	false
-		rawequal(<function>,<function>)	true
-		rawequal(<thread>,<function>)	false
-		rawequal('abc',<thread>)	false
-		rawequal(1.25,<thread>)	false
-		rawequal(true,<thread>)	false
-		rawequal(<table>,<thread>)	false
-		rawequal(<function>,<thread>)	false
-		rawequal(<thread>,<thread>)	true
--- checkallerrors
-		rawequal()	...bad argument...
--- checkallerrors
-		rawequal('abc')	...bad argument...
-		rawequal(1.25)	...bad argument...
-		rawequal(true)	...bad argument...
-		rawequal(<table>)	...bad argument...
-		rawequal(<function>)	...bad argument...
-		rawequal(<thread>)	...bad argument...
====== rawget ======
--- checkallpass
-		rawget(<table>,'aa')	456
--- checkallpass
-		rawget(<table>,'abc')	
-		rawget(<table>,1.25)	
-		rawget(<table>,true)	
-		rawget(<table>,<table>)	
-		rawget(<table>,<function>)	
--- checkallerrors
--- checkallerrors
-		rawget(nil,'abc')	...bad argument...
-		rawget('abc','abc')	...bad argument...
-		rawget(1.25,'abc')	...bad argument...
-		rawget(true,'abc')	...bad argument...
-		rawget(<function>,'abc')	...bad argument...
-		rawget(<thread>,'abc')	...bad argument...
-		rawget(nil,1.25)	...bad argument...
-		rawget('abc',1.25)	...bad argument...
-		rawget(1.25,1.25)	...bad argument...
-		rawget(true,1.25)	...bad argument...
-		rawget(<function>,1.25)	...bad argument...
-		rawget(<thread>,1.25)	...bad argument...
-		rawget(nil,true)	...bad argument...
-		rawget('abc',true)	...bad argument...
-		rawget(1.25,true)	...bad argument...
-		rawget(true,true)	...bad argument...
-		rawget(<function>,true)	...bad argument...
-		rawget(<thread>,true)	...bad argument...
-		rawget(nil,<table>)	...bad argument...
-		rawget('abc',<table>)	...bad argument...
-		rawget(1.25,<table>)	...bad argument...
-		rawget(true,<table>)	...bad argument...
-		rawget(<function>,<table>)	...bad argument...
-		rawget(<thread>,<table>)	...bad argument...
-		rawget(nil,<function>)	...bad argument...
-		rawget('abc',<function>)	...bad argument...
-		rawget(1.25,<function>)	...bad argument...
-		rawget(true,<function>)	...bad argument...
-		rawget(<function>,<function>)	...bad argument...
-		rawget(<thread>,<function>)	...bad argument...
--- checkallerrors
-		rawget()	...bad argument...
====== rawset ======
--- checkallpass
-		rawset(<table>,'aa','abc')	<table>
-		rawset(<table>,'aa',1.25)	<table>
-		rawset(<table>,'aa',true)	<table>
-		rawset(<table>,'aa',<table>)	<table>
-		rawset(<table>,'aa',<function>)	<table>
-		rawset(<table>,'aa',<thread>)	<table>
--- checkallpass
-		rawset(<table>,'abc','abc')	<table>
-		rawset(<table>,1.25,'abc')	<table>
-		rawset(<table>,true,'abc')	<table>
-		rawset(<table>,<table>,'abc')	<table>
-		rawset(<table>,<function>,'abc')	<table>
-		rawset(<table>,'abc',1.25)	<table>
-		rawset(<table>,1.25,1.25)	<table>
-		rawset(<table>,true,1.25)	<table>
-		rawset(<table>,<table>,1.25)	<table>
-		rawset(<table>,<function>,1.25)	<table>
-		rawset(<table>,'abc',true)	<table>
-		rawset(<table>,1.25,true)	<table>
-		rawset(<table>,true,true)	<table>
-		rawset(<table>,<table>,true)	<table>
-		rawset(<table>,<function>,true)	<table>
-		rawset(<table>,'abc',<table>)	<table>
-		rawset(<table>,1.25,<table>)	<table>
-		rawset(<table>,true,<table>)	<table>
-		rawset(<table>,<table>,<table>)	<table>
-		rawset(<table>,<function>,<table>)	<table>
-		rawset(<table>,'abc',<function>)	<table>
-		rawset(<table>,1.25,<function>)	<table>
-		rawset(<table>,true,<function>)	<table>
-		rawset(<table>,<table>,<function>)	<table>
-		rawset(<table>,<function>,<function>)	<table>
-		rawset(<table>,'abc',<thread>)	<table>
-		rawset(<table>,1.25,<thread>)	<table>
-		rawset(<table>,true,<thread>)	<table>
-		rawset(<table>,<table>,<thread>)	<table>
-		rawset(<table>,<function>,<thread>)	<table>
--- checkallerrors
--- checkallerrors
-		rawset()	...bad argument...
--- checkallerrors
-		rawset(nil,'abc','abc')	...bad argument...
-		rawset('abc','abc','abc')	...bad argument...
-		rawset(1.25,'abc','abc')	...bad argument...
-		rawset(true,'abc','abc')	...bad argument...
-		rawset(<function>,'abc','abc')	...bad argument...
-		rawset(<thread>,'abc','abc')	...bad argument...
-		rawset(nil,1.25,'abc')	...bad argument...
-		rawset('abc',1.25,'abc')	...bad argument...
-		rawset(1.25,1.25,'abc')	...bad argument...
-		rawset(true,1.25,'abc')	...bad argument...
-		rawset(<function>,1.25,'abc')	...bad argument...
-		rawset(<thread>,1.25,'abc')	...bad argument...
-		rawset(nil,'abc',1.25)	...bad argument...
-		rawset('abc','abc',1.25)	...bad argument...
-		rawset(1.25,'abc',1.25)	...bad argument...
-		rawset(true,'abc',1.25)	...bad argument...
-		rawset(<function>,'abc',1.25)	...bad argument...
-		rawset(<thread>,'abc',1.25)	...bad argument...
-		rawset(nil,1.25,1.25)	...bad argument...
-		rawset('abc',1.25,1.25)	...bad argument...
-		rawset(1.25,1.25,1.25)	...bad argument...
-		rawset(true,1.25,1.25)	...bad argument...
-		rawset(<function>,1.25,1.25)	...bad argument...
-		rawset(<thread>,1.25,1.25)	...bad argument...
--- checkallerrors
-		rawset(<table>,'aa')	...bad argument...
====== select ======
--- checkallpass
-		select(1.25,nil)	
-		select('#',nil)	1
-		select(1.25,'abc')	'abc'
-		select('#','abc')	1
-		select(1.25,1.25)	1.25
-		select('#',1.25)	1
-		select(1.25,true)	true
-		select('#',true)	1
-		select(1.25,<table>)	<table>
-		select('#',<table>)	1
-		select(1.25,<function>)	<function>
-		select('#',<function>)	1
-		select(1.25,<thread>)	<thread>
-		select('#',<thread>)	1
--- checkallerrors
-		select(nil)	...bad argument...
-		select('abc')	...bad argument...
-		select(true)	...bad argument...
-		select(<table>)	...bad argument...
-		select(<function>)	...bad argument...
-		select(<thread>)	...bad argument...
====== setmetatable ======
--- checkallpass
-		setmetatable(<table>,nil)	<table>
-		setmetatable(<table>,<table>)	<table>
--- checkallerrors
-		setmetatable(nil,<table>)	...bad argument...
-		setmetatable('abc',<table>)	...bad argument...
-		setmetatable(1.25,<table>)	...bad argument...
-		setmetatable(true,<table>)	...bad argument...
-		setmetatable(<function>,<table>)	...bad argument...
-		setmetatable(<thread>,<table>)	...bad argument...
--- checkallerrors
-		setmetatable(<table>,'abc')	...bad argument...
-		setmetatable(<table>,1.25)	...bad argument...
-		setmetatable(<table>,true)	...bad argument...
-		setmetatable(<table>,<function>)	...bad argument...
-		setmetatable(<table>,<thread>)	...bad argument...
====== tonumber ======
--- checkallpass
-		tonumber(1.25,nil)	1.25
-		tonumber('789',nil)	789
-		tonumber(1.25,2)	
-		tonumber('789',2)	
-		tonumber(1.25,10)	
-		tonumber('789',10)	789
-		tonumber(1.25,36)	
-		tonumber('789',36)	9369
--- checkallpass
-		tonumber('abc',nil)	
-		tonumber(1.25,nil)	1.25
-		tonumber(true,nil)	
-		tonumber(<table>,nil)	
-		tonumber(<function>,nil)	
-		tonumber(<thread>,nil)	
--- checkallpass
-		tonumber('abc',10)	
-		tonumber(1.25,10)	
--- checkallerrors
-		tonumber(nil,10)	...bad argument...
-		tonumber(true,10)	...bad argument...
-		tonumber(<table>,10)	...bad argument...
-		tonumber(<function>,10)	...bad argument...
-		tonumber(<thread>,10)	...bad argument...
--- checkallerrors
-		tonumber(nil,2)	...bad argument...
-		tonumber(<function>,2)	...bad argument...
-		tonumber(<table>,2)	...bad argument...
-		tonumber(nil,9)	...bad argument...
-		tonumber(<function>,9)	...bad argument...
-		tonumber(<table>,9)	...bad argument...
-		tonumber(nil,11)	...bad argument...
-		tonumber(<function>,11)	...bad argument...
-		tonumber(<table>,11)	...bad argument...
-		tonumber(nil,36)	...bad argument...
-		tonumber(<function>,36)	...bad argument...
-		tonumber(<table>,36)	...bad argument...
--- checkallerrors
-		tonumber(1.25,1)	...bad argument...
-		tonumber('789',1)	...bad argument...
-		tonumber(1.25,37)	...bad argument...
-		tonumber('789',37)	...bad argument...
-		tonumber(1.25,<table>)	...bad argument...
-		tonumber('789',<table>)	...bad argument...
-		tonumber(1.25,<function>)	...bad argument...
-		tonumber('789',<function>)	...bad argument...
-		tonumber(1.25,true)	...bad argument...
-		tonumber('789',true)	...bad argument...
====== tostring ======
--- checkallpass
-		tostring('abc')	'abc'
-		tostring(1.25)	'1.25'
-		tostring(true)	'true'
--- checkallpass
-		tostring(<table>)	string
-		tostring(<function>)	string
-		tostring(<thread>)	string
--- checkallpass
-		tostring('abc','anchor')	'abc'
-		tostring(1.25,'anchor')	'1.25'
-		tostring(true,'anchor')	'true'
--- checkallpass
-		tostring(<table>,'anchor')	string
-		tostring(<function>,'anchor')	string
-		tostring(<thread>,'anchor')	string
--- checkallerrors
-		tostring()	...bad argument...
====== type ======
--- checkallpass
-		type('abc')	'string'
-		type(1.25)	'number'
-		type(true)	'boolean'
-		type(<table>)	'table'
-		type(<function>)	'function'
-		type(<thread>)	'thread'
--- checkallpass
-		type(nil,'anchor')	'nil'
-		type('abc','anchor')	'string'
-		type(1.25,'anchor')	'number'
-		type(true,'anchor')	'boolean'
-		type(<table>,'anchor')	'table'
-		type(<function>,'anchor')	'function'
-		type(<thread>,'anchor')	'thread'
--- checkallerrors
-		type()	...bad argument...
====== xpcall ======
--- checkallpass
-		xpcall('abc','abc')	boolean,string
-		xpcall(1.25,'abc')	boolean,string
-		xpcall(true,'abc')	boolean,string
-		xpcall(<table>,'abc')	boolean,string
-		xpcall(<function>,'abc')	boolean
-		xpcall(<thread>,'abc')	boolean,string
-		xpcall('abc',1.25)	boolean,string
-		xpcall(1.25,1.25)	boolean,string
-		xpcall(true,1.25)	boolean,string
-		xpcall(<table>,1.25)	boolean,string
-		xpcall(<function>,1.25)	boolean
-		xpcall(<thread>,1.25)	boolean,string
-		xpcall('abc',true)	boolean,string
-		xpcall(1.25,true)	boolean,string
-		xpcall(true,true)	boolean,string
-		xpcall(<table>,true)	boolean,string
-		xpcall(<function>,true)	boolean
-		xpcall(<thread>,true)	boolean,string
-		xpcall('abc',<table>)	boolean,string
-		xpcall(1.25,<table>)	boolean,string
-		xpcall(true,<table>)	boolean,string
-		xpcall(<table>,<table>)	boolean,string
-		xpcall(<function>,<table>)	boolean
-		xpcall(<thread>,<table>)	boolean,string
-		xpcall('abc',<thread>)	boolean,string
-		xpcall(1.25,<thread>)	boolean,string
-		xpcall(true,<thread>)	boolean,string
-		xpcall(<table>,<thread>)	boolean,string
-		xpcall(<function>,<thread>)	boolean
-		xpcall(<thread>,<thread>)	boolean,string
--- checkallpass
-		xpcall('abc',<function>)	boolean,string
-		xpcall(1.25,<function>)	boolean,string
-		xpcall(true,<function>)	boolean,string
-		xpcall(<table>,<function>)	boolean,string
-		xpcall(<function>,<function>)	boolean
-		xpcall(<thread>,<function>)	boolean,string
--- checkallerrors
-		xpcall(nil)	...bad argument...
-		xpcall('abc')	...bad argument...
-		xpcall(1.25)	...bad argument...
-		xpcall(true)	...bad argument...
-		xpcall(<table>)	...bad argument...
-		xpcall(<function>)	...bad argument...
-		xpcall(<thread>)	...bad argument...
